src/meltano/cli/add.py

```diff
@@ -112,10 +116,13 @@ def add(
     if flags["include_related"]:
         related_plugin_types = list(PluginType)
 
-    related_plugins = add_related_plugins(
-        project, plugins, add_service=add_service, plugin_types=related_plugin_types
+    required_plugins = add_required_plugins(
+        project,
+        plugins,
+        add_service=add_service,
+        plugin_types=related_plugin_types,
     )
-    plugins.extend(related_plugins)
+    plugins.extend(required_plugins)
 
     # We will install the plugins in reverse order, since dependencies
```

Comment:
@edgarrmondragon does the new `required:` functionality rely on this reverse order install? It was removed by #5957 as a redundant requirement of the previous `add_related` functionality, but I see here it is tested working with the new `required_plugins` extended list 🤔

Reply:
@kgpayne Nope, it shouldn't rely on it. The diff is a bit weird cause this is definitely not present in `main`